docs/website/docs/plus/getting-started/advanced_tutorial.md

```diff
@@ -0,0 +1,305 @@
+:::
+
+### Writing data back to the catalog
+
+The datasets on the dlt+ catalogs may also be written to. The datasets provide `write` method that can be used to write data back to the catalog. You will be able to control which datasets may be written to with contracts in the future. Under the hood, `dlt+` will use an ad-hoc dlt pipeline to run this operation.
+
+:::warning
+Writing data back to the catalog is an experimental feature at this time and should be used with caution until it is fully stable.
+:::
+
+```py
+# get a dataset from the catalog (this dataset must already exist physically and be explicitely defined in the dlt.yml)
+dataset = current.catalog().dataset("my_pipeline_dataset")
+# write a dataframe to the dataset into the table "my_table"
+dataset.write(pd.DataFrame({"name": ["John", "Jane", "Jim"], "age": [30, 25, 35]}), table_name="my_table")
+```
+
+You can also iterate over dataframes or arrow tables in an existing table and write them to a new table in the same or another dataset:
+
+```py
+# get dataset from the catalog
+dataset = current.catalog().dataset("my_pipeline_dataset")
+
+# here we iterate over the existing items table and write each dataframe to the dataset into a new table
+def transform_frames():
+    # iterate of table `items` in the dataset in chunks of 1000 rows
+    for df in dataset.items.iter_df(chunk_size=1000):
+        # do something with the dataframe here
+        yield df
+
+# write the dataframe to the dataset into the table "my_new_table"
+dataset.write(transform_frames, table_name="my_new_table")
+```
+
+### Switching profiles in code
+
+By default if you access the project, you will be using the default or pinned profile. You can switch the profile with the `switch_profile` function. Consider the following example:
+
+```py   
+from dlt_plus import current
+from dlt_plus.project.run_context import switch_profile
+
+if __name__ == "__main__":
+    # will show that the default profile is active
+    print(current.project().current_profile)
+    # switch to the tests profile
+    switch_profile("tests")
+    # now the tests profile is active and is merged with the project config
+    print(current.project().current_profile)
+```
+
+## Packaging a project
+
+`dlt+` also has tools to help you package a dlt+ project for distribution. This will make your dlt+ project pip installable and make it easier for you to distribute accross your organization. To start with a dlt+ packaged project, you can supply the package argument to the project init command:
+
+```sh
+dlt project init arrow duckdb --package my_dlt_project
+```
+
+This will provide you with the same project as used in the basic tutorial but places in a module directory named `my_dlt_project` and provided with a simple PEP compatible `pyproject.toml` file. You will also receive a default package `__init__.py` file to make the module accssible from the outside when installed:
```

Comment:
```suggestion
This will provide you with the same project as used in the basic project tutorial but places in a module directory named `my_dlt_project` and provided with a simple PEP compatible `pyproject.toml` file. You will also receive a default package `__init__.py` file to make the module accssible from the outside when installed:
```